### DISCUSSION TITLE
Combine bed sheets selection with heightmaps

### DISCUSSION THREAD

**Discussion by togeis** [jschuh/klipper-macros#96] [Ideas]
Hi,

is it possible to combine a bed sheet with it's specific heightmap, so G29 is not to be called on every bed sheet change?

New feature request: Load heightmap for selected bed sheet

Regards
Tobias

**Comment by jschuh** [comment]
Interesting. I suppose I could have `PRINT_START` just check `bed_mesh.profiles` for one with the same name as the current sheet, and if it exists load that instead of running `BED_MESH_CALIBRATE_FAST`.

Of course, that puts the onus on the user to make sure they've saved a full mesh profile at the appropriate temperature. I could add a macro to simplify that, but it's still going to be a bit more error prone than running an optimized mesh on every print.

I'll think about it.

**Comment by jschuh** [comment]
I played around with this a bit yesterday and [got an implementation working here](https://github.com/jschuh/klipper-macros/compare/main...surface_mesh).

**Comment by jschuh** [comment]
Closing since this landed on main in [v0.8.0](https://github.com/jschuh/klipper-macros/releases/tag/v0.8.0).

**Reply by togeis** [reply]
Thx, tested today, works really well !
